cli/tests/program.rs

```diff
@@ -488,13 +488,27 @@ fn test_cli_program_deploy_with_authority() {
         program_data[..]
     );
 
-    // Set a new authority
+    // Set a new authority sign offline first
     let new_upgrade_authority = Keypair::new();
     config.signers = vec![&keypair, &upgrade_authority];
     config.command = CliCommand::Program(ProgramCliCommand::SetUpgradeAuthority {
         program_pubkey,
         upgrade_authority_index: Some(1),
         new_upgrade_authority: Some(new_upgrade_authority.pubkey()),
+        sign_only: true,
+        dump_transaction_message: false,
+    });
+    let sig_response = process_command(&config).unwrap();
+    let sign_only = parse_sign_only_reply_string(&sig_response);
+    let offline_pre_signer = sign_only.presigner_of(&keypair.pubkey()).unwrap();
+
+    config.signers = vec![&keypair, &upgrade_authority];
+    config.command = CliCommand::Program(ProgramCliCommand::SetUpgradeAuthority {
+        program_pubkey,
+        upgrade_authority_index: Some(1),
+        new_upgrade_authority: Some(new_upgrade_authority.pubkey()),
+        sign_only: false,
+        dump_transaction_message: false,
```

Comment:
Same here, needs the blockhash query:
```suggestion
        sign_only: false,
        dump_transaction_message: false,
        blockhash_query: BlockhashQuery::new(Some(blockhash), true, None),
```

Reply:
I think this one the second argument should be `false` instead of `true`